Support named PK constraints (#14999)
crate/crate: docs/sql/general/constraints.rst

Adding a ``PRIMARY KEY`` column is only possible if the table is empty.

Syntax::

  [CONSTRAINT <name>] PRIMARY KEY [ column_name [, ... ] ]

> For example, a table with a named ``PRIMARY KEY`` constraint can be create

Reviewer: ```suggestion
For example, a table with a named ``PRIMARY KEY`` constraint can be created
```

Author: Thank you!

> For example, a table with a named ``PRIMARY KEY`` constraint can be create like

Reviewer: Maybe:
```suggestion
For example, a table with a named ``PRIMARY KEY`` constraint can be created with::
```
and remove `below::`

below::

    cr> CREATE TABLE person (
    ...     firstname TEXT,
    ...     lastname TEXT,
    ...     CONSTRAINT c PRIMARY KEY (firstname, lastname)
    ... );
    CREATE OK, 1 row affected  (... sec)

The named ``PRIMARY KEY`` constraints can be inlined::

    cr> CREATE TABLE person2 (
    ...     firstname TEXT CONSTRAINT c PRIMARY KEY,
    ...     lastname TEXT CONSTRAINT c PRIMARY KEY
    ... );
    CREATE OK, 1 row affected  (... sec)

> If another column is required to be added as ``PRIMARY KEY`` column::

Reviewer: ```suggestion
If a new column is required to be added with ``PRIMARY KEY`` constraint::
```

    cr> ALTER TABLE person2 ADD COLUMN middleName PRIMARY KEY;
    ALTER OK, -1 rows affected  (... sec)

> The ``PRIMARY KEY`` constraint can be unnamed like below::

Reviewer: ```suggestion
The ``PRIMARY KEY`` constraint can also be unnamed, e.g.::
```

Syntax::

  [CONSTRAINT <check_name>] CHECK (boolean_expression)

> If ``CONSTAINT <check_name>`` is omitted, CrateDB generates a unique name

Reviewer: ```suggestion
If ``CONSTRAINT <check_name>`` is omitted, CrateDB generates a unique name
```
